src/provider/interface.ts

```diff
@@ -21,8 +23,8 @@ export type PublicProvider = {
   networkInfos(): Promise<Network>
   getOperationStatus(opId: string): Promise<OperationStatus>
   getEvents(filter: EventFilter): Promise<t.OutputEvents>
-  getNodeStatus(): Promise<NodeStatusInfo>
-  readSC(params: ReadSCParams): Promise<ReadSCData>
+  getNodeStatus(): Promise<NodeStatusInfo | PublicStatus>
+  readSC(params: ReadSCParams): Promise<ReadSCData | ReadOnlyExecutionOutput>
```

Comment:
It will be pretty annoying to have to different return type possible.
It is really needed ?
two solution to this:
- modify `getNodeStatus`and `readSC`so they return `NodeStatusInfo`and `ReadSCData`
- modify `NodeStatusInfo`and `ReadSCData` to be more generic and fit the new needs